aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -0,0 +1,223 @@
+import base64
+import json
+import re
+import zlib
+from enum import Enum
+from typing import Any, Callable, Dict, List, Optional, Set, Tuple, Union
+
+from aws_lambda_powertools.shared.json_encoder import Encoder
+from aws_lambda_powertools.utilities.data_classes import ALBEvent, APIGatewayProxyEvent, APIGatewayProxyEventV2
+from aws_lambda_powertools.utilities.data_classes.common import BaseProxyEvent
+from aws_lambda_powertools.utilities.typing import LambdaContext
+
+
+class ProxyEventType(Enum):
+    http_api_v1 = "APIGatewayProxyEvent"
+    http_api_v2 = "APIGatewayProxyEventV2"
+    alb_event = "ALBEvent"
+    api_gateway = http_api_v1
+
+
+class CORSConfig(object):
+    _REQUIRED_HEADERS = ["Content-Type", "X-Amz-Date", "Authorization", "X-Api-Key", "X-Amz-Security-Token"]
+
+    def __init__(
+        self,
+        allow_origin: str = "*",
```

Comment:
This can be overridden when using `CORSConfig` directly. Otherwise this is by design.

Reply:
That's the only thing I'm nervous about it - I understand people hate CORS for not knowing it, and wanting to have a quick switch but this more often than we'd like end up in a bad place.

I'm happy with the entire implementation except these two things then we can merge, or else this will be flagged by security (and for good reasons), and we'd have to take the feature down - this will impact customers the hard way

1. Allow origin should be a required option. Docs can show * in dev and stress the danger of it. If customers want to use * in prod it's a consent rather than a suboptimal default

2. Allow credentials set to False by default

Reply:
One thing that you can do with Chalice is set the CORS behavior once, and then just have a Boolean flag for which end points you want to have enabled. Should we also allow for that? Maybe for the DX we simplify it to just be something you can set once for the whole lambda. And only toggle on or off at the endpoints

Reply:
Happy with that - have a method to set CORS or any default headers if you'd like so you can reuse it. Then a toggle one